### DISCUSSION TITLE
discontinue package

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#23462] [Suggest an Idea]
### Tell us more.

Currently, I am using renovate for internal packages, which tend to have a more dynamic and often shorter lifespan compared to 3rd party packages.

At the moment, the feature replacementName only handles one option for dealing with discontinuation, which is replacement. However, there are several other options to consider:

1. Complete discontinuation: This means dropping the package entirely. For example, if you have a wrapper around a 3rd party library that should no longer be used.

2. The functionality of the package was divided into 2 or more other packages.

3. The package was branched. For instance, in the case of a Docker image, it might be branched into two images, each containing a different subset of functionality, that should not be needed together.

4. Combining packages: Merging multiple libraries into a single one.

One specific default behavior that this feature can start with is allowing to configure a custom message. This message would be posted as a comment where the discontinued package is being used. The purpose of this message would be to alert the maintainers about the discontinuation of the package and provide instructions of what next.

Of course adder handlings could be added, like adding all the libraries the original was split into, or creating several PRs for developer to choose from for which image to use when branching.

**Comment by rarkins** [comment]
This sounds like the earlier/interrupted work we did on "deprecated" packages.

**Comment by chezsmithy** [comment]
Noting a trend where we are seeing abandoned packages that get forked and the original package never gets released again. Here is an example: https://github.com/oracle/python-cx_Oracle

The package has been replaced but any references to the cx_Oracle package just sit un changed until the repo breaks or appears with unpatched CVEs. 

What might be interesting is if the dependency dashboard could display the last release of a given package and we could flag a warning for packages not released within say 2 years.

**Reply by rarkins** [reply]
I refer to this category as "abandoned" packages

**Reply by HonkingGoose** [reply]
> Noting a trend where we are seeing abandoned packages that get forked and the original package never gets released again. Here is an example: https://github.com/oracle/python-cx_Oracle

We have crowd-sourced replacement presets. Do you want to add a replacement rule for that `oracle/python-cx-Oracle` project?

[Renovate docs, replacement presets](https://docs.renovatebot.com/presets-replacements/)

**Reply by HonkingGoose** [reply]
> What might be interesting is if the dependency dashboard could display the last release of a given package and we could flag a warning for packages not released within say 2 years.

This issue seems related, as it's asking for more detail on the Dependency Dashboard, but for packages that _have a PR_, or _pending update_. So it's not about showing stale packages:

- #17749
